### PR TITLE
Fold in comments to array_map()

### PR DESCRIPTION
**Pull request by Crell**
Also added some types to the examples along the way, because types.

### CODE REVIEW

File: reference/array/functions/array-map.xml [hunk 1]
```diff
@@ -22,7 +22,9 @@
    used as arguments for the callback.
    The number of parameters that the <parameter>callback</parameter>
    function accepts should match the number of arrays
-   passed to <function>array_map</function>.
+   passed to <function>array_map</function>. If it accepts fewer, excess
+   input arrays will be ignored. If it accepts more, an <classname>ArgumentCountError</classname>
+   is thrown.
```
Comment (kamil-tekiela) [inline reference/array/functions/array-map.xml:27]: I'm dumb and it took me 2 minutes to understand this sentence. Could we simplify this for non-native speakers?

How about this?
```suggestion
   passed to <function>array_map</function>. Excess
   input arrays are ignored. An <class>ArgumentCountError</class>
   is thrown if an insufficient number of arguments is provided.
```

Also, the previous sentence uses the present tense, so let's stick to using the present tense.

File: reference/array/functions/array-map.xml [hunk 2]
```diff
@@ -371,6 +378,43 @@ array(1) {
...
   }
 }
+]]>
+    </screen>
+   </example>
+   <example>
+    <title><function>array_map</function> - associative arrays</title>
+    <para>
+     While <function>array_map</function> does not directly support
+     using the array key as an input, that may be simulated using <function>array_keys</function>
```
Comment (kamil-tekiela) [inline reference/array/functions/array-map.xml:388]: Missing period at the end. Why not use `this` instead of `that`?

Comment (Crell) [reply reference/array/functions/array-map.xml:388]: As a native speaker, "this" feels clumsier in context.  Both are grammatically valid, but "this" sounds worse to my American ear.

File: reference/array/functions/array-map.xml (hunk 1 above)
Comment (Crell) [reply reference/array/functions/array-map.xml:27]: I wouldn't call you dumb. :smile:  I'm not sure why it's unclear, though.  It's a series of if-then statements.  That should be self-evident to developers, no?  And it's already in present tense...

Comment (cmb69) [reply reference/array/functions/array-map.xml:27]: At least there is a mismatch between "will be ignored" and "is thrown". :)

Comment (kamil-tekiela) [reply reference/array/functions/array-map.xml]: Oops seems my suggestion clashed with your initial forced push. Class -> classname

File: reference/array/functions/array-map.xml [hunk 3]
```diff
@@ -371,6 +378,43 @@ array(1) {
...
+    <programlisting role="php">
+<![CDATA[
+<?php
+$arr = [
+    'v1' => 'First release',
+    'v2' => 'Second release',
+    'v3' => 'Third release',
+];
+
+$callback = fn(string $k, string $v): string => "$k was the $v";
```
Comment (cmb69) [inline reference/array/functions/array-map.xml:399]: Maybe either use an anonymous function, or note that arrow functions are only available as of PHP 7.4.0.

Comment (Crell) [reply reference/array/functions/array-map.xml:399]: Arrow functions are anonymous functions. :smile: A previous example already offers both versions (also updated in this PR), and within a few months all supported language versions will include short lambdas.  Is there a need to repeat it?

Comment (cmb69) [reply reference/array/functions/array-map.xml:399]: The manual is still supposed to document PHP 7.0.0 and up, and most of these versions don't understand arrow functions. IMO, a short comment makes sense.

Comment (Crell) [reply reference/array/functions/array-map.xml:400]: Comment added.

**Review by cmb69 — APPROVED** [review]
Thank you!

**Pull request merged**